agent/dns.go

```diff
@@ -1457,16 +1464,16 @@ func (d *DNSServer) handleRecurse(resp dns.ResponseWriter, req *dns.Msg) {
 }
 
 // resolveCNAME is used to recursively resolve CNAME records
-func (d *DNSServer) resolveCNAME(name string) []dns.RR {
+func (d *DNSServer) resolveCNAME(name string, maxRecursionLevel int) []dns.RR {
 	// If the CNAME record points to a Consul address, resolve it internally
 	// Convert query to lowercase because DNS is case insensitive; d.domain is
 	// already converted
-	if strings.HasSuffix(strings.ToLower(name), "."+d.domain) {
+	if strings.HasSuffix(strings.ToLower(name), "."+d.domain) && maxRecursionLevel > 0 {
```

Comment:
`.consul` domains will fallback to recursors when max recursion level is reached, we probably don't want that

```suggestion
	if strings.HasSuffix(strings.ToLower(name), "."+d.domain) {
	    if maxRecursionLevel <= 0 {
	        d.logger.Println("[ERR] dns: max recursion level %d reached when trying to resolve %s, stopping", maxRecursionLevel, name)
	        return nil
	    }
```

Reply:
Yes, DONE